### PR TITLE
RavenDB-19209 To check that query relay on `PropertyNameConverter` to get property name

### PR DESCRIPTION
**Pull request by maciejaszyk**
Original PR: https://github.com/ravendb/ravendb/pull/14819

### Issue link
https://issues.hibernatingrhinos.com/issue/RavenDB-19209

### Additional description
This merges some renaming behavior deep in the guts of the LINQ provider to enable user extensions.

### Type of change
* Bug fix

### How risky is the change?
* High

I merged behavior across duplicated `if` branches, we'll need to ensure that no test is broken, but I don't expect it to.

### Backward compatibility
* breaking change

### Is it platform specific issue?
* No

### Documentation update
* No documentation update is needed

### Testing
* Tests have been added that prove the fix is effective or that the feature works

### CODE REVIEW

**Comment by marcuslindblom** [comment]
When will this be available on MyGet?

File: test/SlowTests/Issues/RavenDB-11089.cs [hunk 1]
```diff
@@ -433,14 +431,14 @@ public void Can_Project_Into_Class_With_Let(RavenTestParameters config)
...
-}
-from 'Users' as user select output(user)", queryAsString);
+//                     RavenTestHelper.AssertEqualRespectingNewLines(
+//                         @"declare function output(user) {
+//     var first = user.name;
+//     var last = user.lastName;
+//     var format = function(){return first+"" ""+last;};
+//     return { FullName : format() };
+// }
+// from 'Users' as user select output(user)", queryAsString);
```
Comment (arekpalinski) [inline test/SlowTests/Issues/RavenDB-11089.cs]: Why is this commented?

File: src/Raven.Server/Documents/Indexes/Index.cs [hunk 2]
```diff
@@ -3813,24 +3813,24 @@ private void AssertQueryDoesNotContainFieldsThatAreNotIndexed(QueryMetadata meta
...
 
-            ThrowInvalidField(f, indexName);
+            ThrowInvalidField(f, queryMetadata);
         }
 
-        private static void ThrowInvalidField(string f, string indexName)
+        private static void ThrowInvalidField(string f, QueryMetadata queryMetadata)
         {
-            throw new ArgumentException($"The field '{f}' is not indexed for index '{indexName}', cannot query/sort on fields that are not indexed");
+            throw new ArgumentException($"The field '{f}' is not indexed, cannot query/sort on fields that are not indexed in query: " + queryMetadata.QueryText);
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/Indexes/Index.cs]: Why did you change the index name to query text here? The index name was suitable I think

File: src/Raven.Client/Documents/Conventions/DocumentConventions.cs [hunk 3]
```diff
@@ -376,6 +376,20 @@ public Version HttpVersion
         }
 #endif
 
+        public string GetConvertedPropertyNameFor(MemberInfo member)
```
Comment (arekpalinski) [inline src/Raven.Client/Documents/Conventions/DocumentConventions.cs]: Can it be `internal`?

File: src/Raven.Server/Documents/Indexes/Index.cs (hunk 2 above)
Comment (maciejaszyk) [reply src/Raven.Server/Documents/Indexes/Index.cs]: It's @ayende change.

File: test/SlowTests/Issues/RavenDB-11089.cs (hunk 1 above)
Comment (maciejaszyk) [reply test/SlowTests/Issues/RavenDB-11089.cs]: Reverted.

File: src/Raven.Client/Documents/Conventions/DocumentConventions.cs (hunk 3 above)
Comment (maciejaszyk) [reply src/Raven.Client/Documents/Conventions/DocumentConventions.cs]: yes, done

File: src/Raven.Server/Documents/Indexes/Index.cs (hunk 2 above)
Comment (arekpalinski) [reply src/Raven.Server/Documents/Indexes/Index.cs]: Let's have both - index name (as it was) and the query.

**Review by arekpalinski — APPROVED** [review]
